### PR TITLE
App rating

### PR DESCRIPTION
**Pull request by nwingt**

### CODE REVIEW

File: app/models/root-store/setup-root-store.ts [hunk 1]
```diff
@@ -53,6 +53,18 @@ function createRootStore(env: Environment, data: any = {}) {
   env.authCoreAPI.callbacks.unauthorized = rootStore.handleUnauthenticatedError
   env.likeCoAPI.config.onUnauthenticated = rootStore.handleUnauthenticatedError
   env.likerLandAPI.config.onUnauthenticated = rootStore.handleUnauthenticatedError
+
+  // Reset app rating cooldown at first open
+  if (!rootStore.userStore.appRatingCooldown) {
+    rootStore.userStore.resetAppRatingCooldown()
+  } else if (
+    rootStore.userStore.ratedAppVersion &&
+    rootStore.userStore.ratedAppVersion !== env.appConfig.getValue("APP_VERSION")
```
Comment (williamchong007) [inline app/models/root-store/setup-root-store.ts:62]: Maybe we should judge major version changes (and maybe even more than a few major version changes) for less disturbing UX

**Comment by nwingt** [comment]
@williamchong007 Please review again

**Comment by williamchong007** [comment]
The new behaviour is that client shows the rating prompt after cooldown, client storea a list of prompted version in store to prevent reprompting. Then to trigger new rating prompts, we update the app rating version in config?
It feels like we don't need to be storing a list, just do numeric comparison on rating version will do?

**Comment by williamchong007** [comment]
Also if the rating config is updated via remote config, it might cause this value to be outsync with actual version
Should we store the current app version instead, and do semver comparison with rating's target app version?

**Comment by nwingt** [comment]
@williamchong007 Updated

File: app/i18n/en.json [hunk 2]
```diff
@@ -1,4 +1,8 @@
 {
+  "AppRatingPrompt": {
+    "Title": "Do you satisfy with Liker Land?",
```
Comment (williamchong007) [inline app/i18n/en.json:3]: `Are you satisfied with Liker Land`

**Pull request merged**